### ISSUE TITLE
Emit events for tabline extension

### ISSUE THREAD

**Issue by michaelbrusegard** [xarvex/presentation.wez#1]
Would it be possible to emit events for when the presentation modes is opened and closed? This way it will be possible to create an extension for [tabline.wez](https://github.com/michaelbrusegard/tabline.wez), for customising the tab-bar for presentation mode

**Comment by xarvex** [comment]
Interesting idea! I'm sure I can put something together real quick to take care of that.
On that separate note, that's a very cool tab you made, I'll be checking it out!

**Comment by xarvex** [comment]
Would just like to update that I am still looking at this, just busy with university semester starting up. Expect a push sometime towards the weekend. I have not forgotten!

**Comment by michaelbrusegard** [comment]
In case it helps, here is the link to the relevant API I think :) https://wezfurlong.org/wezterm/config/lua/wezterm/emit.html
There is no rush though, I am also in uni so I understand the ack of time

**Comment by xarvex** [comment]
Sorry for the delay, I finally found the time to get around to this. I'm cranking it out now, currently troubleshooting the actual firing of the events.
I've also cloned your tabline to make the extension, I want to be sure what I implement is a working solution.

**Comment by xarvex** [comment]
Implemented in https://github.com/xarvex/presentation.wez/commit/55f4bbfe17d3273a2347d6e04903fcecd3f2ee11 and made a PR to tabline.wez at https://github.com/michaelbrusegard/tabline.wez/pull/20. Keeping this open for now just to make sure all is satisfactory.

Note so you do not experience the same pain I did: it turns out the Lua debug in Wezterm does not listen to events, so do not use that to test and go crazy when nothing is being listened to. :sweat_smile:
